### PR TITLE
ci(edge-release): update existing instead of re-create

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:

To avoid spamming our fellow coders GitHub timelines, update the
existing edge release with new binaries and a new SHA target instead
of creating a new edge release every time.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Walther — APPROVED** [review]
looks good to me 👍
and as discussed, noting that the docker image tag needs to be manually pushed before merging

File: images/circleci-release/garden.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Module
+type: container
+name: circleci-release
```
Comment (edvald) [inline images/circleci-release/garden.yml:3]: Might I suggest just using the `circleci-runner` for this? Seems like unnecessary overhead to add a while new image. Also curious why we can't use an upstream image to begin with?

Comment (stefreak) [reply images/circleci-release/garden.yml:3]: Why a custom image is necessary: If we update the existing release, we need another utility called `gh` that is not included together with `ghr` in any upstream container I looked at.

I can change this to circleci-runner if you prefer

Comment (edvald) [reply images/circleci-release/garden.yml:3]: Aha I see, makes sense. But yeah, I think just the one CircleCI image makes most sense.

Comment (stefreak) [reply images/circleci-release/garden.yml:3]: OK, I'll change that

Comment (stefreak) [reply images/circleci-release/garden.yml]: @edvald OK, changed it now. But to be able to upload the image I need permission on dockerhub

Comment (stefreak) [reply images/circleci-release/garden.yml]: CI ran with the new image now, feel free to approve and then I'd just merge this and we could look at whether this actually works

**Comment by edvald** [comment]
Can't say if it's related, but the `test-windows` check seems to fail reliably. Could you take a look at the @stefreak? We'll want to fix it whether it's related to these changes or not.

**Comment by stefreak** [comment]
@edvald this test also fails on main: https://app.circleci.com/pipelines/github/garden-io/garden/14309/workflows/e38043f4-eb14-4713-aa31-06edbc541e70/jobs/242005

**Comment by Walther** [comment]
> Can't say if it's related, but the `test-windows` check seems to fail reliably. Could you take a look at the @stefreak? We'll want to fix it whether it's related to these changes or not.

gcloud auth plugin component missing, PR here https://github.com/garden-io/garden/pull/3519

**Comment by edvald** [comment]
Yep all good! PR looks solid, just wanted to check for awareness on that.

Sent via Superhuman iOS ( ***@***.*** )

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**